tests/test_security_path_traversal.py

```
@@ -204,3 +204,101 @@ def test_run_tailer_path_validation(tmp_path):
+                await apply_evolution_patch_endpoint(req)
+            assert exc.value.status_code == 400
+            assert "run_id" in str(exc.value.detail)
+
+            req = ApplyEvolutionRequest(patch_id=f"valid_run:{bad_id}", dry_run=True, create_backup=True)
+            with pytest.raises(HTTPException) as exc:
+                await apply_evolution_patch_endpoint(req)
+            assert exc.value.status_code == 400
+            assert "patch_id" in str(exc.value.detail)
+
+        # Better test for apply_evolution_patch_endpoint without composite id
+        def dummy_get_evo_module():
```

Ruff (0.15.2)

[warning] Missing return type annotation for private function `dummy_get_evo_module`

(ANN202)